pyogrio/_io.pyx

```diff
@@ -1031,6 +1031,11 @@ def ogr_read(
     if sql is not None and layer is not None:
         raise ValueError("'sql' paramater cannot be combined with 'layer'")
 
+    if not read_geometry and not return_fids and columns is not None and len(columns) == 0:
```

Comment:
```suggestion
    if not (read_geometry or return_fids or (columns is not None and len(columns) > 0)):
```

(slight simplification)

Reply:
columns being None is also OK, so changed to:
``` python
if not (read_geometry or return_fids or columns is None or len(columns) > 0):
```